### ISSUE TITLE
Problems in printing sentence split time

### ISSUE THREAD

**Issue by ampli** [opencog/link-grammar#1404]
``` text
$ link-parser -v=2
link-grammar: Info: Dictionary found at ./data/en/4.0.dict
link-grammar: Info: Dictionary version 5.12.1, locale en_US.UTF-8
link-grammar: Info: Library version link-grammar-5.12.1. Enter "!help" for help.
linkparser> x
#### Finished tokenizing (3 tokens)
++++ Split sentence                              0.10 seconds
++++ Finished expression pruning                 0.00 seconds
++++ Built disjuncts                             0.00 seconds
++++ Eliminated duplicate disjuncts              0.00 seconds
++++ Encoded for pruning (skipped)               0.00 seconds
++++ power pruned (for 0 nulls)                  0.00 seconds
++++ pp pruning                                  0.00 seconds
++++ Encoded for parsing                         0.00 seconds
++++ Initialized fast matcher                    0.00 seconds
++++ Counted parses (1 w/0 nulls)                0.00 seconds
++++ Built parse set                             0.00 seconds
++++ Postprocessed all linkages                  0.00 seconds
++++ Sorted all linkages                         0.00 seconds
++++ Finished parse                              0.00 seconds
++++ Time                                        0.00 seconds (0.00 total)
Found 1 linkage (1 had no P.P. violations)
	Unique linkage, cost vector = (UNUSED=0 DIS= 0.00 LEN=1)

    +->Wa--+
    |      |
LEFT-WALL x.n

linkparser> x
#### Finished tokenizing (3 tokens)
++++ Split sentence                              0.00 seconds
++++ Finished expression pruning                 0.00 seconds
++++ Built disjuncts                             0.00 seconds
++++ Eliminated duplicate disjuncts              0.00 seconds
++++ Encoded for pruning (skipped)               0.00 seconds
++++ power pruned (for 0 nulls)                  0.00 seconds
++++ pp pruning                                  0.00 seconds
++++ Encoded for parsing                         0.00 seconds
++++ Initialized fast matcher                    0.00 seconds
++++ Counted parses (1 w/0 nulls)                0.00 seconds
++++ Built parse set                             0.00 seconds
++++ Postprocessed all linkages                  0.00 seconds
++++ Sorted all linkages                         0.00 seconds
++++ Finished parse                              0.00 seconds
++++ Time                                        0.00 seconds (0.00 total)
```

As you can see above, the first sentence split time printout includes the setup time of the dict (and the calling program, like `link-parser` or its equivalent).
This is because there is no `resources_reset()` at the start of `sentence_split()`.
https://github.com/opencog/link-grammar/blob/7951ed835929a5737d185dbfc223d4770a2d7dd3/link-grammar/api.c#L506-L514
The following sentence split time printouts also may have an added time since the start point of the measurement is not fixed (it includes the processing done between sentence parsings, which can be arbitrary and relatively big).

However, adding `resources_reset()` at the start of `sentence_split()` is also problematic. This is because the `+++ Time` line including the total, will not be the sum of the displayed times, as it would not include the split time.
Trying to fix that by removing `resources_reset()` in sentence parse would introduce a bug: Repeated parsings (e.g. due to 2-step parsing in which the 0-nulls is done first and then parsing for nulls, or panic parsing, would include incorrect timing as the time will not reset before the repeated parsing.

My fix proposal is to do one of these:
1. Remove this added `print_time(opts, "Split sentence"); `, as sentence splits are very short, and debugging their improvements needs precise CPU profiling anyway.
2. Add `resources_reset()` at the start of `sentence_split()` and change `+++ Time` to `+++ Time w/o sentence split` to make clear that it doesn't include sentence split.
In that case, I would also remove the `### Finish tokenizing` message and change `print_time(opts, "Split sentence")` to `print_time(opts, "Split sentence (%zu tokens)\n", sent->length);`.

**Comment by linas** [comment]
> sentence splits are very short, 

Heh. Not at all! They are short only if the dictionary is fast. If the dictionary is slow, then this measures the time needed to look up all of the words. This can be 5 seconds for some of my dictionaries, and half-a-minute, if I am connected over the network (AtomSpace runs over the network, and my current dictionary format can result in megabytes of stuff serialized across sockets.)

As to "how to fix this", I don't care. I suppose it can be removed; I was just using it to track what was happening during debugging.  Now, it would be *nice* to have an accurate printout, but ... I look at this only very rarely.

FWIW, the core bug is that it was taking a minute to parse a sentence; but the total time was reported as 20 seconds, and it took me a while before I realized the rest of the time was lost looking up dictionary words. So that `print_time()` was an attempt to fix that, but I guess I botched it.
